### ISSUE TITLE
[Feature Request] Flatten Nested Objects

### ISSUE THREAD

**Issue by Genae** [don-tnowe/godot-resources-as-sheets-plugin#73]
First of all: Thanks for the great plugin!
I have a feature Request. These are my resources:
```
public partial class StatBlockRes: Resource
{
    [Export] public GrowingStat MaxHP;
    [Export] public GrowingStat MaxMP;
    [Export] public GrowingStat PAtk;
    [Export] public GrowingStat MAtk;
    [Export] public GrowingStat PDef;
    [Export] public GrowingStat MDef;
    [Export] public GrowingStat Crit;
    [Export] public GrowingStat ArmorPen;
}

public partial class GrowingStat: Resource
{
    [Export] public float Value;
    [Export] public float Growth;
}
```

Now what I would love to see is a table with the following headers:
`MaxHP.Value | MaxHP.Growth | MaxMP.Value | MaxMP.Growth ....`

Would be amazing if it also worked on arrays, maybe something along the lines of
`Array.0.Value | Array.0.Growth | Array.1.Value | Array.1.Growth`

**Comment by don-tnowe** [comment]
Good feature.

It's already possible to open a property as a table: click a column header's arrow, then `Open Subresources of Column` to edit them as if you'd opened a folder of resources.

Maybe I can get it part-way by supporting `to_string()` overrides for built-in resources, so the values can be at least viewed in context of the full table.

**Comment by Genae** [comment]
Yes, opening a property as a table is what I am doing now, but especially for stat blocks it is important to see the whole picture if you want to compare them against each other.
I was thinking of just putting MaxHP_Value and MaxHP_Growth directly into the statblock but it just doesn't feel right :)

I have seen other people having issues with importing nested objects through the csv import, this could be a solution for that problem as well. (At least if the object is not to wild on nested arrays)

**Comment by don-tnowe** [comment]
Did this in 133e34ea57b240bdf576ca0642720f9c13fa4ede, with one "but": scripts of resources inside your property must have the `@tool` attribute at the beginning of the file so that `_to_string` gets called.

It is because a script function is being called, so in the editor their functions won't be compiled, which is also the case for custom setters.

There is [a workaround](https://github.com/don-tnowe/godot-inspector-extender/blob/f2b30372cccfa7e566912c86fa934ae268d12e58/addons/inspector_extender/inspector_plugin.gd#L108) but with too many resources this might hit performance.

**Comment by Genae** [comment]
I am going to test this tomorrow and give you feedback

**Comment by Genae** [comment]
hmm, i have overwritten the ToString() properties of both of my resources:
<img width="290" alt="image" src="https://github.com/user-attachments/assets/78a1e0fe-93a5-4d66-8705-7640866c7470" />
<img width="458" alt="image" src="https://github.com/user-attachments/assets/460f7a34-5199-4f5b-a434-ecd783ca2814" />
but:
- It seems to have no impact on the display, maybe the c# ToString is not the same as the gd script _to_string
<img width="746" alt="image" src="https://github.com/user-attachments/assets/00645210-5a77-451a-a946-040449eb7553" />

**Comment by Genae** [comment]
hmm, looking at the godot documentation there currently seems to be no c# equivalent to that:
<img width="690" alt="image" src="https://github.com/user-attachments/assets/0e504d8a-3af8-42dc-91a1-8a452946d002" />

**Comment by don-tnowe** [comment]
It is a thing in C# - the `override` is just a method on every C# object, not Godot-specific. Just that it seems I forgot that C# exists and has a different naming convention (*despite your original code snippet being in C#, oops!*)

So, now I pushed it to the main branch. The label's style is configurable in the Settings menu (*top-right corner*). Don't forget the `[Tool]` attribute, I almost concluded that it wouldn't work at all. Hope I'm not causing too much inconvenience!

**Comment by Genae** [comment]
Well, I know the override in C#, I've been coding C# for more than 10 years now :D I am however new to godot and gdscript. I did actually spot the code where you check for the method name and replaced it with the C# version but it didn't work.

Now adding the Tool attribute is a thing I wasn't aware of so it now actually works! It is amazing and I love it! You are not causing any inconvenience, all of this has been a great experience and the result is spot on ;)

Thanks!
<img width="452" alt="image" src="https://github.com/user-attachments/assets/7c2116f6-3ca6-40f7-94f7-cb102b3d8f93" />

I will keep this issue opened in case you still want to do the whole flattening thing. The readonly version however is already amazing and well worth while.
